src/Form/Form.php

```diff
@@ -274,7 +274,7 @@ public function execute(array $data, array $options = []): bool
      * @param array $data Form data.
      * @return bool
      */
-    protected function _execute(array $data): bool
+    protected function process(array $data): bool
```

Comment:
This will have to be explicitly mentioned in the migration docs as this method is meant to be overridden by user forms.

Reply:
that is my plan after this is all merged

Reply:
If we wanted to, we could deprecated this in cake 5 and check if _execute() is declared in a derived class or not.

Reply:
Is `Form::_execute()` the only method we want to deprecate this way or do want to do that for all the other methods as well? Meaning:
- add `process()` method in `5.next`
- move current `_execute()` method contents to `process()` and call `process()` inside current `_execute()`
- add deprecation warning to `_execut()`

Reply:
Any with extension point (likely to be existing on app/plugin side).
Afaik that only fits this one method, right?

Reply:
At least for these methods changed in here I don't see any as well.
But there are (probably) others which just have a removed `_` so we can do that for those as well.

Reply:
We could always do a `method_exists` and call the `_execute()` method if it exists for backwards compatibility. It would also give us a spot to trigger a deprecation.

Reply:
It's already been handled for 5.next :)
https://github.com/cakephp/cakephp/pull/18725